subsys/bootloader/provision/provision_flash.c

```diff
@@ -37,34 +48,57 @@ u32_t num_public_keys_read(void)
 	return p_provision_data->num_public_keys;
 }
 
+static const u32_t *get_key(u32_t key_idx)
+{
+	return &(p_provision_data->key_data[key_idx *
+			(CONFIG_SB_PUBLIC_KEY_HASH_LEN / 4 + 1)]);
+}
+
+#define INVALID_VAL 0xFFFF0000
+
+static const u32_t *is_valid(u32_t key_idx)
+{
+	return get_key(key_idx);
+}
+
 int public_key_data_read(u32_t key_idx, u32_t *p_buf, size_t buf_size)
 {
 	const u32_t *p_key;
 
+	if (*is_valid(key_idx) == INVALID_VAL) {
```

Comment:
Why do we need the indirection of `is_valid` here? To me it looks better to have `if(*get_key() == INVALID_VAL)`

Reply:
Refactored this a bit. Removed `is_valid()` and added an offset argument to `get_key()`.